### PR TITLE
WebGLShadowMap: Simplify morph targets check.

### PR DESCRIPTION
**Pull request by Mugen87**
The `updateMorphTargets()` methods of `Mesh` and `Points` only support `BufferGeometry` since a while now. Hence, I think it's okay to remove some legacy code in `WebGLShadowMap`.

### CODE REVIEW

**Comment by mrdoob** [comment]
Just double checking. This removes support for `Geometry` with morph targets in `WebGLShadowMap`, right?

**Comment by Mugen87** [comment]
Exactly.

**Comment by mrdoob** [comment]
Hmm... Oh! Is it because there is no way a `Geometry` can reach here because by this point we have already converted it to `BufferGeometry`?

**Comment by Mugen87** [comment]
> Is it because there is no way a Geometry can reach here because by this point we have already converted it to BufferGeometry?

Up to now, `WebGLShadowMap` always used the geometry object from `object.geometry` (which could be of type `THREE.Geometry`).

However, now that you have mentioned it....it's actually not hard to use the converted instance of `BufferGeometry` which should make the PR more robust! Please check out the last commit and things should become clear.

File: src/renderers/webgl/WebGLShadowMap.js [hunk 1]
```diff
@@ -297,17 +295,9 @@ function WebGLShadowMap( _renderer, _objects, maxTextureSize ) {
 
 			var useMorphing = false;
 
-			if ( material.morphTargets === true ) {
-
-				if ( geometry.isBufferGeometry === true ) {
-
-					useMorphing = geometry.morphAttributes && geometry.morphAttributes.position && geometry.morphAttributes.position.length > 0;
+			if ( material.morphTargets === true && geometry.isBufferGeometry === true ) {
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLShadowMap.js:298]: Do we need the ` && geometry.isBufferGeometry === true` check?

Comment (Mugen87) [reply src/renderers/webgl/WebGLShadowMap.js]: Not anymore 😊 🎉

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**